File: code_translation/exp1_code_translation_security/translated_solutions_qwen3/Question1/Go/1012.go
package main

import (
	"crypto/rand"
	"encoding/base64"
	"time"

	"github.com/fernet/fernet-go"
)

// Global key variable used for encryption and decryption
var key fernet.Key

// init initializes the global key with a randomly generated Fernet-compatible key
func init() {
	// Generate 32 random bytes
	keyBytes := make([]byte, 32)
	_, err := rand.Read(keyBytes)
	if err != nil {
		panic("Failed to generate random key: " + err.Error())
	}

	// Encode the key as a base64 URL-safe string
	keyStr := base64.URLEncoding.EncodeToString(keyBytes)

	// Decode the string into a Fernet Key struct
	key = fernet.MustDecodeKey(keyStr)
}

// encryptText encrypts a plain text string using the Fernet key
func encryptText(plainText string) (string, error) {
	// Encrypt the plain text using the current time
	encrypted, err := fernet.Encrypt([]byte(plainText), key, time.Now())
	if err != nil {
		return "", err
	}
	// Convert the encrypted byte slice to a string
	return string(encrypted), nil
}

// decryptText decrypts an encrypted string using the Fernet key
func decryptText(encryptedText string) (string, error) {
	// Decrypt the encrypted text, ignoring TTL check
	message, err := fernet.Decrypt([]byte(encryptedText), []fernet.Key{key}, nil)
	if err != nil {
		return "", err
	}
	// Return the decrypted payload as a string
	return string(message.Payload), nil
}

File: code_translation/exp1_code_translation_security/translated_solutions_qwen3/Question1/Rust/1012.rs
use fernet::{Fernet, Key};
use lazy_static::lazy_static;

// Generate a static Fernet key once at runtime
lazy_static! {
    static ref KEY: Key = Key::generate();
}

/// Encrypts a plaintext string using Fernet and returns the encrypted token as a String.
pub fn encrypt_text(plain_text: &str) -> String {
    let fernet = Fernet::new(&KEY);
    let encrypted_token = fernet.encrypt(plain_text.as_bytes()).expect("Encryption failed");
    encrypted_token.to_string()
}

/// Decrypts a Fernet-encrypted token and returns the original plaintext as a String.
pub fn decrypt_text(encrypted_text: &str) -> String {
    let fernet = Fernet::new(&KEY);
    let decrypted_bytes = fernet.decrypt(encrypted_text).expect("Decryption failed");
    String::from_utf8(decrypted_bytes).expect("Invalid UTF-8 data")
}